tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -0,0 +1,68 @@
+import cv2
+import numpy as np
+import torch
+import torchvision
+from torchvision.models._utils import IntermediateLayerGetter
+
+
+class DFBRegistrtation:
+    r"""Deep Feature based Registration
+
+    This class implements a CNN feature based registration,
+    as proposed in a paper titled `Deep Feature based Cross-slide Registration
+    <https://arxiv.org/pdf/2202.09971.pdf>`_.
+
+    """
+
+    def __init__(self):
+        self.patch_size = (224, 224)
+        self.Xscale, self.Yscale = [], []
+        model = torchvision.models.vgg16(True)
+        return_layers = {"16": "block3_pool", "23": "block4_pool", "30": "block5_pool"}
+        self.FeatureExtractor = IntermediateLayerGetter(
+            model.features, return_layers=return_layers
+        )
+
+    def extract_features(self, fixed_img, moving_img):
```

Comment:
Does this function needs to be exposed? It can be made private when the full pipeline is implemented. At this stage making it private will make it difficult to run tests.

Reply:
Yes, we can make it private later.

Reply:
Thanks. Please add the suggestion above and put the comment so this can be followed up later.

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/b718c77ef780c6930b720f3b8a37e130aea1f6b4/tiatoolbox/tools/registration/wsi_registration.py#L28-L29